fuel-tx/src/transaction.rs

```diff
@@ -451,6 +453,81 @@ impl From<Mint> for Transaction {
     }
 }
 
+impl Serialize for Transaction {
```

Comment:
What's the benefit of implementing this kind of boilerplate manually? Does it just add unnecessary complexity to the macros by trying to automate this?

Reply:
We can use the `peek` function, which removes the `Clone` requirement from the `trait Input: Clone`. Skipping of the discriminant is a very "magic" thing and I prefer to do that manually because we need it only in one place for `Transaction`.